hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java

```diff
@@ -903,40 +904,63 @@ private void registerFeed(String tenantId) throws Exception {
             // rest of the URL says we want the feeds API
             url.append("feeds");
 
-            // now send the REST request
-            OkHttpClient httpclient = this.httpClientBuilder.getHttpClient();
-            Request request = this.httpClientBuilder.buildJsonPostRequest(url.toString(),
-                    Collections.singletonMap("Hawkular-Tenant", tenantId), jsonPayload);
-            Response httpResponse = httpclient.newCall(request).execute();
-
-            // HTTP status of 201 means success; 409 means it already exists, anything else is an error
-            if (httpResponse.code() == 201) {
-
-                // success - store our feed ID so we remember it the next time
-                final String feedObjectFromServer = httpResponse.body().string();
-                final Feed feed = Util.fromJson(feedObjectFromServer, Feed.class);
-                if (desiredFeedId.equals(feed.getId())) {
-                    log.infoUsingFeedId(feed.getId());
-                } else {
-                    log.errorUnwantedFeedId(feed.getId(), desiredFeedId);
-                    // should we throw an error here or just use the feed ID we were given?
-                    log.debugf("Using feed ID [%s]; make sure the agent doesn't lose its data file", feed.getId());
+            // because we can support multiple tenants, we need to register our feed under all tenants
+            List<AbstractEndpointConfiguration> endpoints = new ArrayList<>();
+            endpoints.addAll(configuration.getDmrConfiguration().getEndpoints().values());
+            endpoints.addAll(configuration.getJmxConfiguration().getEndpoints().values());
+            endpoints.addAll(configuration.getPlatformConfiguration().getEndpoints().values());
+            endpoints.addAll(configuration.getPrometheusConfiguration().getEndpoints().values());
+
+            List<String> tenantIds = new ArrayList<String>();
```

Comment:
OK, I just figured why make multiple calls if 10 managed servers all used the same tenantId.  if that's not an issue then go ahead and merge.